src/datasets/arrow_reader.py

```diff
@@ -19,23 +19,18 @@
 import math
 import os
 import re
-import shutil
 from dataclasses import dataclass
 from functools import partial
-from pathlib import Path
 from typing import TYPE_CHECKING, List, Optional, Union
 
 import pyarrow as pa
 import pyarrow.parquet as pq
 from tqdm.contrib.concurrent import thread_map
 
-from .download.download_config import DownloadConfig
```

Comment:
The removal of this code line makes apparent an underlying circular import: https://github.com/huggingface/datasets/actions/runs/9694146568/job/26751140360?pr=6996
```
    import datasets
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/__init__.py:17: in <module>
    from .arrow_dataset import Dataset
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/arrow_dataset.py:76: in <module>
    from .arrow_reader import ArrowReader
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/arrow_reader.py:31: in <module>
    from .table import InMemoryTable, MemoryMappedTable, Table, concat_tables
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/table.py:13: in <module>
    from .utils.logging import get_logger
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/utils/__init__.py:17: in <module>
    from .info_utils import VerificationMode
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/utils/info_utils.py:8: in <module>
    from ..exceptions import (
/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/exceptions.py:8: in <module>
    from .table import CastError
E   ImportError: cannot import name 'CastError' from partially initialized module 'datasets.table' (most likely due to a circular import) (/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/datasets/table.py)
```

Reply:
I have reverted the removal of this unused import, so that the underlying circular import remain hidden.

I would propose addressing the underlying circular import in a subsequent specific PR.